src/Features/CSharp/Portable/ExtractMethod/CSharpSelectionValidator.cs

```diff
@@ -72,18 +74,66 @@ internal partial class CSharpSelectionValidator : SelectionValidator
                 }
             }
 
+            // Warn if local functions are in selection since data flow analysis
+            // cannot correctly analyze them
+            // https://github.com/dotnet/roslyn/issues/14214
+            if (SpanContainsLocalFunction(selectionInfo.FinalSpan, model, root))
```

Comment:
At this point, i would say "SpanInvolvesLocalFunction" :)  Contains has fairly specific meaning with span APIs.  So the implementation is unexpected.  If you leave it vague with 'Involves' then it makes more sense what's going on.